### PR TITLE
[3.x] Allow pinning property values + Consistent property defaults (nicer version)

### PR DESCRIPTION
**Pull request by RandomShaper**
Version of #52943 for 3.x.

Supersedes #52234.

~**NOTE:** CI build issues are unrelated to the changes in this PR.~ **UPDATE:** Fixed. It seems `override` is taboo in 3.x.

### CODE REVIEW

File: doc/classes/EditorProperty.xml [hunk 1]
```diff
@@ -120,6 +120,14 @@
 				Emit it if you want to key a property with a single value.
 			</description>
 		</signal>
+		<signal name="property_pinned">
+			<argument index="0" name="property" type="String" />
+			<argument index="1" name="pinned" type="bool" />
+			<description>
+				Emit it if you want to mark (or unmark) a the value of a property for being saved regardless being equal to the default.
```
Comment (akien-mga) [inline doc/classes/EditorProperty.xml:127]:
```suggestion
				Emit it if you want to mark (or unmark) the value of a property for being saved regardless of being equal to the default.
```

File: doc/classes/PackedScene.xml [hunk 2]
```diff
@@ -86,5 +86,9 @@
 			If passed to [method instance], provides local scene resources to the local scene. Only the main scene should receive the main edit state.
 			[b]Note:[/b] Only available in editor builds.
 		</constant>
+		<constant name="GEN_EDIT_STATE_MAIN_INHERITED" value="3" enum="GenEditState">
+			It's similar to [code]GEN_EDIT_STATE_MAIN[/code], but for the case where the scene is being instantiated to be the base of another one.
```
Comment (akien-mga) [inline doc/classes/PackedScene.xml:90]:
```suggestion
			It's similar to [constant GEN_EDIT_STATE_MAIN], but for the case where the scene is being instantiated to be the base of another one.
```

File: doc/classes/SceneState.xml [hunk 3]
```diff
@@ -168,5 +168,9 @@
 			If passed to [method PackedScene.instance], provides local scene resources to the local scene. Only the main scene should receive the main edit state.
 			[b]Note:[/b] Only available in editor builds.
 		</constant>
+		<constant name="GEN_EDIT_STATE_MAIN_INHERITED" value="3" enum="GenEditState">
+			If passed to [method PackedScene.instance], it's similar to [code]GEN_EDIT_STATE_MAIN[/code], but for the case where the scene is being instantiated to be the base of another one.
```
Comment (akien-mga) [inline doc/classes/SceneState.xml:172]:
```suggestion
			If passed to [method PackedScene.instance], it's similar to [constant GEN_EDIT_STATE_MAIN], but for the case where the scene is being instantiated to be the base of another one.
```

Comment (RandomShaper) [reply doc/classes/SceneState.xml:172]: Coming in next push.

File: doc/classes/PackedScene.xml (hunk 2 above)
Comment (RandomShaper) [reply doc/classes/PackedScene.xml:90]: Coming in next push.

File: doc/classes/EditorProperty.xml (hunk 1 above)
Comment (RandomShaper) [reply doc/classes/EditorProperty.xml:127]: Coming in next push.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**